lib/elephant_in_the_room/sites/post.ex

```diff
@@ -156,21 +163,40 @@ defmodule ElephantInTheRoom.Sites.Post do
   end
 
   def get_or_insert_tag(name, site_id) do
-    inserted_tag = Repo.insert(%Tag{name: name, site_id: site_id, color: "686868"},
-      [on_conflict: :nothing])
+    inserted_tag =
+      Repo.insert(
+        %Tag{name: name, site_id: site_id, color: "686868"},
+        on_conflict: :nothing
+      )
+
     case inserted_tag do
-      %{id: id} when id != nil -> inserted_tag
+      %{id: id} when id != nil ->
+        inserted_tag
+
       _ ->
         Repo.get_by!(Tag, name: name, site_id: site_id)
     end
   end
 
-  def generate_og_meta(conn, %Post{title: title, thumbnail: image, abstract: description} = post) do
+  defp parse_date(%{"date" => date, "hour" => hour} = attrs) do
+    [h, m, s] =
+      hour
+      |> Enum.map(fn {_, v} -> if String.length(v) == 1, do: "0" <> v, else: v end)
+
+    iso_hour = %{hour: h, minute: m, second: s}
+    date_string = "#{date} #{iso_hour.hour}:#{iso_hour.minute}:#{iso_hour.second}"
+    {:ok, datetime} = NaiveDateTime.from_iso8601(date_string)
```

Comment:
Maybe `NaiveDateTime.new` can do it better https://hexdocs.pm/elixir/NaiveDateTime.html#new/8

Reply:
I did it that way because the date and hour comes as a string. So to use `NaiveDateTime.new` I have convert each string to integer.